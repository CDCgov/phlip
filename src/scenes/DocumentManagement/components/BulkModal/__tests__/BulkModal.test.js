import React from 'react'
import { shallow } from 'enzyme'
import { BulkModal } from '../index'

const props = {
  suggestions: [],
  searchValue: '',
  onClearSuggestions: jest.fn(),
  onGetSuggestions: jest.fn(),
  onSearchValueChange: jest.fn(),
  onSuggestionSelected: jest.fn(),
  bulkType: 'approve',
  docCount: 1,
  onCloseModal: jest.fn(),
  open: true,
  buttonInfo: { inProgress: false, diabled: false },
  onConfirmAction: jest.fn()
}

describe('DocumentManagement - BulkModal', () => {
  test('should render correctly', () => {
    expect(shallow(<BulkModal {...props} />)).toMatchSnapshot()
  })
})
