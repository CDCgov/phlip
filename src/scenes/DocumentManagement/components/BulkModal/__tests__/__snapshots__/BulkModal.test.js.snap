// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentManagement - BulkModal should render correctly 1`] = `
<WithStyles(Modal)
  hideOverflow={false}
  id="bulkConfirmBox"
  maxWidth="md"
  onClose={[MockFunction]}
  open={true}
>
  <ModalTitle
    title="Approve"
  />
  <WithStyles(Divider) />
  <ModalContent
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 250,
        "paddingTop": 24,
        "width": 500,
      }
    }
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={true}
      justify="space-between"
      padding={0}
      raised={false}
      style={Object {}}
      type="column"
    >
      <FlexGrid
        align="stretch"
        circular={false}
        container={false}
        flex={false}
        justify="stretch"
        padding={0}
        raised={false}
        style={Object {}}
        type="column"
      >
        <WithStyles(Typography)
          align="center"
          variant="body1"
        >
          Number of Documents Selected: 
          1
        </WithStyles(Typography)>
      </FlexGrid>
    </FlexGrid>
  </ModalContent>
  <WithStyles(Divider) />
  <WithStyles(ModalActions)
    actions={
      Array [
        Object {
          "onClick": [MockFunction],
          "otherProps": Object {
            "aria-label": "Close modal",
          },
          "preferred": true,
          "type": "button",
          "value": "Cancel",
        },
        Object {
          "disabled": undefined,
          "onClick": [MockFunction],
          "otherProps": Object {
            "aria-label": "Confirm",
            "id": "bulkConfirmBtn",
          },
          "type": "button",
          "value": <React.Fragment>
            Confirm
          </React.Fragment>,
        },
      ]
    }
  />
</WithStyles(Modal)>
`;
